src/array.rs

```diff
@@ -358,12 +358,46 @@ impl<T, D> PyArray<T, D> {
     }
 }
 
-struct InvertedAxises(Vec<Axis>);
+enum InvertedAxes {
+    Short(u32),
+    Long(Vec<usize>),
+}
+
+impl InvertedAxes {
+    fn new(len: usize) -> Self {
+        if len < 32 {
+            Self::Short(0)
+        } else {
+            Self::Long(Vec::new())
```

Comment:
Is it worth perhaps adding a test with a 40-dimension array, to exercise this code path?

Reply:
I tried and actually can't as NumPy seems to be limited to at most 32 dimensions:

```python
In [12]: a = np.zeros(sum(((1,) for _ in range(32)), ()) + (3,))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [12], in <module>
----> 1 a = np.zeros(sum(((1,) for _ in range(32)), ()) + (3,))

ValueError: maximum supported dimension for an ndarray is 32, found 33
```

So it would seem that I do not need the `Long` representation at all?